### DISCUSSION TITLE
Breadcrumbs - Can we turn them off?

### DISCUSSION THREAD

**Discussion by LaocheXe** [e107inc/e107#5059] [Q&A Plugins]
I've notice lately that each page I create for a plugin, there is Breadcrumbs, if the page is broken by my code, you can see {BREADCRUMB} shortcode. But when the code works, you get House Icon / PageName

![Screenshot 2023-08-12 192031](https://github.com/e107inc/e107/assets/5818022/dc62b2b8-e3ff-457a-aa70-3cb51dddff0d)

Now this displays on each page, you would think you would require the shortcode in the page if you wanted to use it, only thing I require is class2.php, unless this is a bootstrap 5 theme shortcode I don't know of? I don't see anything that is in theme_shortcodes.php - so question is, how can I turn it off from displaying in my plugin pages? It would be nice to just slap some code in to disable that on each page that I don't want it on.

I went looking into the Issues section, and found two open issues, but nothing that points to an on/off switch.

#4224 
#3878

**Comment by LaocheXe** [comment]
While looking deeper into bootstrap 5 theme, notice the {---BREADCRUMB---} sections in the layouts (which I didn't think would be in the html files), commenting them out removes the breadcrumbs - is there a code to turn it off on plugin pages without having to edit the theme files?

**Reply by Jimmi08** [reply]
Did you try in your plugin
 
`e107::breadcrumb($breadcrumb_array);`

and set $breadcrumb_array as emty array or NULL?   If empty array is not working, then you create issue to have this fixes. 
 
e107::breadcrumb() is way how to set shortcodes for plugin, so there should be way how to set it off. 

{---BREADCRUMB---} is magic shortcode  and it is theme developer area.  If plugin supports it, it will be displayed, if not, nothing is displayed.  And yes, it should be in layout if you want the same visual for all site.

Those issues above are not related - cpages use non standard way for breadcrumbs and second one is about templating.
